elide-core/src/main/java/com/yahoo/elide/audit/LogMessage.java

```diff
@@ -120,6 +121,19 @@ public String getMessage() {
                 ctx.setVariable(name, expression);
                 singleElementContext.setVariable(name, singleElementExpression);
             }
+
+            final User user = getUser();
+            if (user != null) {
+                final Object opaqueUser = user.getOpaqueUser();
```

Comment:
Let's move the this line and the null check into your getUser function.  Then I think we are good to go.